allspice/utils/list_components.py

```diff
@@ -791,3 +830,81 @@ def _filter_blank_components(
         final_components.append(component)
 
     return final_components
+
+
+def _find_schdoc_file_by_filename_in_ext_repos(
+    design_reuse_repos: list[Repository] | None,
+    prjpcb_file: str,
+    schdoc_file: str,
+    use_first_found_design_reuse_match: bool | None,
+) -> tuple[Repository | None, str]:
+    """
+    Find a matching schematic document in one or more external repositories
+
+    :param design_reuse_repos: A list of external repositories to search
+    :param prjpcb_file: The Altium project file for the design in hand
+    :param schdoc_file: Desired .SchDoc file to find in external repositories
+    :param use_first_found_design_reuse_match: Option to return the file from the first matched repo
+
+    :returns: The first matching external repository containing the target file,
+    and the target filepath of the desired .SchDoc file in that repository.
+    Duplicates found will result in a ValueError.
+    """
+
+    # Get the target POSIX filepath
+    target_filepath = _resolve_prjpcb_relative_path(schdoc_file, prjpcb_file)
+
+    # Get the filepath from the repo root down
+    exact_match_repo_name, exact_match_filepath_without_repo_root = (
+        _get_schdoc_filepath_from_repo_root(target_filepath)
+    )
+    # Get the schdoc filename
+    schdoc_filename = os.path.basename(target_filepath)
```

Comment:
Use `pathlib` here as it is already imported. `os` can be tricky as we're dealing with Windows paths on a Linux machine.